### ISSUE TITLE
Chrome binary can't be found, even though chromium is installed.

### ISSUE THREAD

**Issue by JustifySC** [Betalord/BHBot#39]
<2019/03/18 00:36:45> BHBot v33 started.
<2019/03/18 00:36:45> Idle mode started, no actions will be taken
<2019/03/18 00:36:45> Driver started succesfully
org.openqa.selenium.WebDriverException: unknown error: no chrome binary at C:\Users\L\AppData\Local\Chromium\Application\chrome.exe
  (Driver info: chromedriver=2.44.609538 (b655c5a60b0b544917107a59d4153d4bf78e1b90),platform=Windows NT 10.0.17134 x86_64) (WARNING: The server did not provide any stacktrace information)
Command duration or timeout: 44 milliseconds
Build info: version: 'unknown', revision: 'unknown', time: 'unknown'
System info: host: 'DESKTOP-MLHEQOO', ip: '192.168.56.1', os.name: 'Windows 10', os.arch: 'amd64', os.version: '10.0', java.version: '1.8.0_201'
Driver info: driver.version: RemoteWebDriver
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source)
        at java.lang.reflect.Constructor.newInstance(Unknown Source)
        at org.openqa.selenium.remote.ErrorHandler.createThrowable(ErrorHandler.java:215)
        at org.openqa.selenium.remote.ErrorHandler.throwIfResponseFailed(ErrorHandler.java:167)
        at org.openqa.selenium.remote.JsonWireProtocolResponse.lambda$new$0(JsonWireProtocolResponse.java:53)
        at org.openqa.selenium.remote.JsonWireProtocolResponse.lambda$getResponseFunction$2(JsonWireProtocolResponse.java:91)
        at org.openqa.selenium.remote.ProtocolHandshake.lambda$createSession$22(ProtocolHandshake.java:365)
        at java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
        at java.util.Spliterators$ArraySpliterator.tryAdvance(Unknown Source)
        at java.util.stream.ReferencePipeline.forEachWithCancel(Unknown Source)
        at java.util.stream.AbstractPipeline.copyIntoWithCancel(Unknown Source)
        at java.util.stream.AbstractPipeline.copyInto(Unknown Source)
        at java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
        at java.util.stream.FindOps$FindOp.evaluateSequential(Unknown Source)
        at java.util.stream.AbstractPipeline.evaluate(Unknown Source)
        at java.util.stream.ReferencePipeline.findFirst(Unknown Source)
        at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:368)
        at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:159)
        at org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:142)
        at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:637)
        at org.openqa.selenium.remote.RemoteWebDriver.startSession(RemoteWebDriver.java:250)
        at org.openqa.selenium.remote.RemoteWebDriver.startSession(RemoteWebDriver.java:236)
        at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:137)
        at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:174)
        at MainThread.connectDriver(MainThread.java:720)
        at MainThread.restart(MainThread.java:838)
        at MainThread.run(MainThread.java:989)
        at java.lang.Thread.run(Unknown Source)

Chromium is installed, and chrome.exe is in this directory. Is there anything I can do to fix this error?

**Comment by bbicudo** [comment]
Try uninstalling chromium (remove all data under **C:\users\<username>\appdata\Local\Chromium)**. Make sure nothing is left from the old setup. 

Also wipe **C:\users\<username>\appdata\Local\Temp** directory.

Restart your machine.

Install it once again from the wiki given link (https://commondatastorage.googleapis.com/chromium-browser-snapshots/index.html?prefix=Win_x64/576753/).

Try again (i had same issue on the first setup and this worked for me).

**Comment by JustifySC** [comment]
Just tried the steps you provided and still same error. anything else I can try?

**Comment by JustifySC** [comment]
`<2019/03/18 11:54:47> BHBot v33 started.
<2019/03/18 11:54:51> Idle mode started, no actions will be taken
<2019/03/18 11:54:51> Driver started succesfully
org.openqa.selenium.WebDriverException: unknown error: no chrome binary at C:\Users\L\AppData\Local\Chromium\Application\chrome.exe
  (Driver info: chromedriver=2.44.609538 (b655c5a60b0b544917107a59d4153d4bf78e1b90),platform=Windows NT 10.0.17134 x86_64) (WARNING: The server did not provide any stacktrace information)
Command duration or timeout: 47 milliseconds
Build info: version: 'unknown', revision: 'unknown', time: 'unknown'
System info: host: 'DESKTOP-MLHEQOO', ip: '192.168.56.1', os.name: 'Windows 10', os.arch: 'amd64', os.version: '10.0', java.version: '1.8.0_201'
Driver info: driver.version: RemoteWebDriver
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source)
        at java.lang.reflect.Constructor.newInstance(Unknown Source)
        at org.openqa.selenium.remote.ErrorHandler.createThrowable(ErrorHandler.java:215)
        at org.openqa.selenium.remote.ErrorHandler.throwIfResponseFailed(ErrorHandler.java:167)
        at org.openqa.selenium.remote.JsonWireProtocolResponse.lambda$new$0(JsonWireProtocolResponse.java:53)
        at org.openqa.selenium.remote.JsonWireProtocolResponse.lambda$getResponseFunction$2(JsonWireProtocolResponse.java:91)
        at org.openqa.selenium.remote.ProtocolHandshake.lambda$createSession$22(ProtocolHandshake.java:365)
        at java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
        at java.util.Spliterators$ArraySpliterator.tryAdvance(Unknown Source)
        at java.util.stream.ReferencePipeline.forEachWithCancel(Unknown Source)
        at java.util.stream.AbstractPipeline.copyIntoWithCancel(Unknown Source)
        at java.util.stream.AbstractPipeline.copyInto(Unknown Source)
        at java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
        at java.util.stream.FindOps$FindOp.evaluateSequential(Unknown Source)
        at java.util.stream.AbstractPipeline.evaluate(Unknown Source)
        at java.util.stream.ReferencePipeline.findFirst(Unknown Source)
        at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:368)
        at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:159)
        at org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:142)
        at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:637)
        at org.openqa.selenium.remote.RemoteWebDriver.startSession(RemoteWebDriver.java:250)
        at org.openqa.selenium.remote.RemoteWebDriver.startSession(RemoteWebDriver.java:236)
        at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:137)
        at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:174)
        at MainThread.connectDriver(MainThread.java:720)
        at MainThread.restart(MainThread.java:838)
        at MainThread.run(MainThread.java:989)
        at java.lang.Thread.run(Unknown Source)
<2019/03/18 11:54:52> Something went wrong with driver restart. Will retry in a few minutes... (sleeping)









`

**Comment by ilpersi** [comment]
Can you verify that Chromium is present in the path  C:\Users\L\AppData\Local\Chromium\Application\chrome.exe?

**Comment by ilpersi** [comment]
After reviewing the code, I can confirm that this is happening when Chromium is not installed correctly. Please be sure to follow the [instructions on the wiki](https://github.com/Betalord/BHBot/wiki). I personally recommend to uninstall it and do a clean install with the [mini_installer](https://www.googleapis.com/download/storage/v1/b/chromium-browser-snapshots/o/Win_x64%2F576753%2Fmini_installer.exe?generation=1532053199306713&alt=media).

I will make sure to have a better failsafe code for this kind of circumstances.

**Comment by JustifySC** [comment]
Hey. Sorry I didn't follow up on this sooner. I have identifed the problem and fixed it. The issue was that I changed my local pc username a few months ago, but your application found the old username somehow. I simply created a folder as the older username and I installed chromium in that directory and it worked. Many thanks for your assistance above, it's fully fixed on my end and the bot is working now.


Thanks!
